### DISCUSSION TITLE
How to save plot to PNG?

### DISCUSSION THREAD

**Discussion by sandboiii** [kernc/backtesting.py#405] [Q&A]
Hello, I've used backtesting.py to evaluate my strategy and now I want to save the plot into PNG image to insert it into a report. And this seems to be impossible because bokeh explicitly requires plot object to make any operations with it unlike matplotlib. And on top of this bt.plot() does not return the plot object.  
I really miss plt.savefig(), is there any way to save plot to png? I'm looking for solution which can be used in pipeline without manual interactions.

**Comment by kernc** [comment]
You can use [Save Tool :floppy_disk: ](https://docs.bokeh.org/en/latest/docs/user_guide/tools.html#savetool) on the plot toolbar to export a PNG snapshot manually.
Otherwise, you can use [`Backtest.plot(filename=...)`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Backtest.plot) to grab the whole interactive HTML.

**Reply by sandboiii** [reply]
There are two problems with Save Tool: it saves each graph separately and needs a manual press and confirmations. Interactive html is good, but it can't be inserted into a document. I wish Backtest.plot() would return a plot object, so it could be saved as PNG with bokeh's export_png() function

**Reply by kernc** [reply]
Ah, yes, issue: https://github.com/bokeh/bokeh/issues/8531. :laughing: 

> I wish Backtest.plot() would return a plot object, so it could be saved as PNG with bokeh's export_png() function

Can you open that as a pull request?

**Reply by sandboiii** [reply]
Yeah sure, I'm gonna send pull request soon

**Reply by holychan** [reply]
Hi @sandboiii @kernc 
is there a way to save the plot as PNG programatically?

**Reply by kernc** [reply]
@holychan [`Backtest.plot()`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Backtest.plot) returns a figure object which you can use [`bokeh.io.export.export_png`](https://docs.bokeh.org/en/latest/docs/reference/io.html#bokeh.io.export.export_png) on.
